src/libraries/System.Numerics.Vectors/tests/Matrix4x4Tests.cs

```diff
@@ -819,11 +819,57 @@ public void Matrix4x4CreateReflectionTest01()
                     Vector3 v = point - pp;
                     float d = Vector3.Dot(v, plane.Normal);
                     Vector3 vp = point - 2.0f * d * plane.Normal;
-                    Assert.True(MathHelper.Equal(rp, vp), "Matrix4x4.Reflection did not provide expected value.");
+
+                    Assert.Equal(vp.M11, rp.M11, 0.0f);
```

Comment:
These should be `AssertExtensions.Equal` so we're using https://github.com/dotnet/runtime/blob/7c1f78fb071ec28a7ba2f9e64efa041e5350215d/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs#L931, which explicitly handles concepts like `NaN`, `NegativeInfinity`, `NegativeZero`, and other concepts in a way that ensures better determinism/correctness

Reply:
Done.